src/metabase/core.clj

```diff
@@ -122,6 +123,9 @@
       ;; otherwise update if appropriate
       (sample-data/update-sample-database-if-needed!)))
 
+  (when (slack/slack-configured?)
+    (future (slack/refresh-cache!)))
```

Comment:
If I spin up 4 instances is this going to trigger four simultaneous refreshes? Would it be better to schedule a Quartz job to fire immediately so it only runs on one instance?

Reply:
As mentioned previously, I'm not sure this specific code here makes sense -- it doesn't do anything on first launch _before_ Slack is configured, and it doesn't do much besides that other than triggering one-off updates when an instance is restarted. If an instance runs for a few months, this isn't enough on its own to keep things up to date on its own

Reply:
The specific problem I am trying to solve with this piece is for users who have slack configured today. When they upgrade their slack users/convo cache will be empty and that will cause the `/form_input` route to act strangely. 

Granted the endpoint will fix itself once the route is hit + (time it takes to crawl their slack data), but until then they'll get no results. Would like to hear if you have another fix for that case.

I considered using the Quartz scheduler, I think you're right that that's a better way to handle this.